src/Compilers/Core/Portable/SourceGeneration/AdditionalSourcesCollection.cs

```diff
@@ -53,14 +53,14 @@ public void Add(string hintName, SourceText source)
                     && c != '{'
                     && c != '}')
                 {
-                    throw new ArgumentException(string.Format(CodeAnalysisResources.HintNameInvalidChar, c, i), nameof(hintName));
+                    throw new ArgumentException(string.Format(CodeAnalysisResources.HintNameInvalidChar, hintName, c, i), nameof(hintName));
```

Comment:
>string.Format(CodeAnalysisResources.HintNameInvalidChar, hintName, c, i) [](http://example.com/codeflow?start=48&length=72)

Are we testing this `ArgumentException` and the one below, and calling `.Message` to verify there are sufficient argument to match the format?

Reply:
Added to the `AdditionalSourcesCollectionTests` to validate we're passing the message in.